utils/utils.go

```diff
@@ -0,0 +1,51 @@
+package utils
+
+import (
+	"regexp"
+	"time"
+	"strconv"
+  "log"
+  "github.com/microsoft/BladeMonRT/logging"
+)
+
+/** Class that contains utilities used in BladeMonRT classes. */
+type Utils struct {
+	logger *log.Logger
+}
+
+func NewUtils() *Utils {
+	var logger *log.Logger = logging.LoggerFactory{}.ConstructLogger("Utils")
+	return &Utils{logger : logger}
+}
+
+/** Class that represents an event parsed from event XML. */
+type EventFromXML struct {
+  Provider string
+  EventID int
+  TimeCreated time.Time
+  EventRecordID int
+}
+
+/** Parses out the event `Provider`, `EventID`, TimeCreated(`SystemTime`), `EventRecordID` (which is different from event ID) from the event XML. */
+func (utils *Utils) ParseEventXML(eventXML string) EventFromXML {
+  re := regexp.MustCompile(`.*Provider *Name=[\"\']([^\"]+)[\"\'].*<EventID[^>]*>([0-9]+)</EventID>.*<TimeCreated +SystemTime=[\"\']([0-9\-]*)T.*<EventRecordID>([0-9]+)</EventRecordID>.*`)
```

Comment:
Can we use an XML parser here rather than regex?

Reply:
<Event xmlns="http://schemas.microsoft.com/win/2004/08/events/event">
- <System>
  <Provider Name="CpuSpeedMonitoring" /> 
  <EventID Qualifiers="0">999</EventID> 

I tried this the problem is we wont know what is at "System" at compile time. Golang requires we predefine the XML structure. Would using a library to parse windows event log instead be a good idea?

Reply:
Yeah, let's definitely use a library for this. If Go doesn't support dynamic XML parsing then we can use some XPath library for GO

Reply:
I found one here:

https://www.golangprograms.com/golang-dynamic-xml-parser.html